thunder/core/transform_common.py

```diff
@@ -111,6 +111,49 @@ def check(inp, log_str):
             check(copy_to_out, "output")
 
 
+def _replace_copy_output_with_source(extrace: Trace) -> None:
+    """This is based on the sharp edge of nvfuser's `add_ouput(output, input)` interface,
+    it makes sure that the output of `prims.copy_` is not used as input for any of its subsequent operators in a nvFusion fused operator
+
+    Anti-pattern:
+
+    .. code-block:: python
+
+        [t2] = nvFusion0(x, y)
+            # result = prims.mul(x, y)
+            # a = prims.copy_(result, x)
+            # t2 = prims.add(a, y)
+
+    Do not use `a` after it has been updated, use the `copy_from` variable `result` instead to reflect the dependency:
```

Comment:
I don't see the dependency issue there. In currently handling of nvfuser.copy_, the new output is done via a copy. https://github.com/Lightning-AI/lightning-thunder/blob/dbf6bad30cf38606f2feaec629f19357bba0b74d/thunder/executors/nvfuserex_impl.py#L2317-L2319

Can you point me to a repro where that's giving us issues?

Reply:
It's true that running the peft benchmark will result in `_inplace_copy_sanity_check` raising an error (and that is true too of the smaller examples in `test_update_aliases` that are changed in this pull request).  However, I don't have a clear understanding of why this check checks both the `copy_to` and the output.  The [issue](https://github.com/Lightning-AI/lightning-thunder/issues/265) for adding this check mentions `copy_to` but the [PR](https://github.com/Lightning-AI/lightning-thunder/pull/285) also checks the output without further explanation.  @jjsjann123 is this check on the output unnecessary?

Reply:
A small example that fails without this change is:
```
import torch
import thunder

def h(x):
    x.add_(1).sin().mul_(5)
    return x

a = torch.randn(5, device="cuda")

jfn = thunder.jit(h, skip_inplace_alias_updates=False, skip_inplace_functionalization=True)

actual = jfn(a)
```
with error
```
NotImplementedError: t2 = prims.sin(t1)  # t2: "cuda:0 f32[5]" trying to use <TensorProxy(name="t1", dtype=thunder.dtypes.float32, shape=(5,))> (the output of 'prims.copy_') as input, which is not safe. There is a risk of accessing the wrong memory. If you are sure you don't want to use this check, it can be disabled by setting `disable_inplace_copy_check=True` in `thunder.jit`.
```

Reply:
Summarizing an outside discussion, changes made in NvFuser since the addition of the sanity check mean that the check can be relaxed and the swapping of output and source is no longer necessary.